lib/child_process.js

```diff
@@ -277,7 +271,7 @@ exports.execFile = function(file /*, args, options, callback*/) {
         if (!encoding)
           _stderr.push(chunk);
         else
-          _stderr += chunk;
+          _stderr += chunk.toString(encoding);
```

Comment:
@jasnell The problem this PR is trying to solve is that the units for `maxBuffer` are in bytes, but once you set up a string decoder, there is no way to know how many bytes went into making the string that gets emitted on `data`. That means you can no longer rely on `chunk.length` in case you have multi-byte characters.